Sources/ChessKit/Board.swift

```diff
@@ -22,6 +22,9 @@ public struct Board: Sendable {
     /// The current position represented on the board.
     public var position: Position
 
+    /// All the positions occurred during the game
+    public var gamePositions: [Int: Int]
```

Comment:
I would rename this to `positionHashCounts` or something like that so it's more clear.

I would also make it `private` so it's not exposed to other users since it's more of an internal thing.